src/Microsoft.TestPlatform.CommunicationUtilities/TestRequestSender.cs

```diff
@@ -76,23 +87,51 @@ public void Close()
         }
 
         /// <inheritdoc/>
+        public bool CheckVersionWithTestHost()
+        {
+            var success = false;
+            this.communicationManager.SendMessage(MessageType.VersionCheck, payload: this.highestNegotiatedVersion);
+
+            var message = this.communicationManager.ReceiveMessage();
+
+            if (message.MessageType == MessageType.VersionCheck)
+            {
+                var protocolVersion = this.dataSerializer.DeserializePayload<int>(message);
+
+                // TODO: Should we check if this is valid ?
+                this.highestNegotiatedVersion = protocolVersion;
+                success = true;
+            }
+            else if (message.MessageType == MessageType.ProtocolError)
+            {
+                EqtTrace.Error("TestRequestSender: VersionCheck Failed");
```

Comment:
Protocol.Error doesn't specify what's the highest version supported at the other end? How does user find why version check failed?